File: android/app/src/main/java/jagtheora/misc/SimplePeer.java
package jagtheora.misc;


public abstract class SimplePeer {

    public long peer;

    protected void init() {}

    protected abstract void clear();

    public final boolean b() {
        return this.peer == 0L;
    }

    public void setPeer(long arg0) {
        this.peer = arg0;
    }

    public final void cleanUp() {
        if (!this.b()) {
            this.clear();
        }
    }

    @Override
    public final void finalize() throws Throwable {
        if (!this.b()) {
            this.cleanUp();
        }
        super.finalize();
    }
    public final void a() { this.cleanUp(); }
}

File: src/jvmMain/kotlin/jagtheora/misc/SimplePeer.kt
package jagtheora.misc


abstract actual class SimplePeer {
    /*private*/ actual var peer: Long = 0

    /*protected*/ actual abstract fun clear()

    actual fun a() {
        if (!this.b()) {
            this.clear()
        }
    }

    @Throws(Throwable::class)
    actual fun finalize() {
        if (!this.b()) {
            this.a()
        }
//        super.finalize()
    }

    actual fun b(): Boolean {
        return this.peer == 0L
    }

    companion actual object {
        init {
            init()
        }

        /*protected*/ actual external fun init()
    }
}
